libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1625,5 +1617,18 @@ constexpr char const *ToString(BlockStatus status)
   }
 }
 
+BasicBloomFilter &MainChain::SelectBloomFilter(chain::TransactionLayout const &tx_layout) const
+{
+  Block::Index const index     = tx_layout.valid_from() / 1000000;
```

Comment:
This actually divides transactions into continuous 'generations' one million blocks each. Close yet distinct, in terms of block number, transactions map to the same Bloom bucket.
Would it possibly make sense to better separate them?

Reply:
A Bloom filter is inherently probabilistic: there isn't a scheme that would guard us completely from 'close' transactions sharing a bucket. That said, this design has a bigger problem, so I'm marking it as WIP again for the moment. I may be able to do something about bucketing along the way.